rclpy/rclpy/node.py

```diff
@@ -577,9 +577,18 @@ def get_parameter(self, name: str) -> Parameter:
             undeclared parameters are allowed.
```

Comment:
`get_parameter_or()` should return the `alternative` value if the parameter is uninitialized, it doesn't seem to be doing that.

Reply:
Should it? The docs state it will return the alternative value if the parameter is not *declared*. We could update the docs though if we want to change the behavior.

I hadn't considered this case for `rclcpp` either. I'll go back and check if we need to do anything differently.

Reply:
After playing around with some examples, I agree that it we should return the alternative value for uninitialized parameters. I've handled this case and added a test in 38f85c3

Reply:
FWIW, I've in rclcpp if a static or dynamic parameter is NOT_SET then the `get_parameters_or` always returns the alternative value. I think that's fair, so I've updated this PR to match that behavior: 763ef63

No changes are required in rclcpp, but I'll open up a follow-up PR adding tests for completeness.

Reply:
Sounds good to me